### ISSUE TITLE
a draggable marker should be draggable via the keyboard

### ISSUE THREAD

**Issue by andrewharvey** [maplibre/maplibre-gl-js#8020]
When you add a draggable marker to a map

```js
new maplibregl.Marker({draggable: true})
        .setLngLat(...)
        .addTo(map);
```

by default it should be keyboard focusable (via a `tabindex`) and when it has focus the arrow keys should move the marker allowing users to perform the same behaviour via the keyboard as they currently do via the mouse.

This was originally mentioned at #7790

**Comment by HarelM** [comment]
Thanks for taking the time to open this issue.
Feel free to open a PR to fix this, should be failrly easy nowadays with some AI assistance.

**Comment by andrewharvey** [comment]
> Thanks for taking the time to open this issue.
> Feel free to open a PR to fix this, should be failrly easy nowadays with some AI assistance.

To be honest it's demotivating to contribute when 1 hour after creating this issue an AI generated PR has been created by someone (or a bot) with seemingly no connection GL JS, and while I can't find any faults with the PRs, I kind of feel like PRs should be created by humans with a connection to the software, not someone shooting AI PR darts out across Github just to score points.

Our design decisions should be carefully considered and UX changes like this and the other PR should be road tested by a human to get a feel for the changes, beyond just unit tests, and any impacts to developers building with the library and to users should be considered.

**Comment by HarelM** [comment]
@andrewharvey I share your pain, especially since I'm reviewing all these PRs.
Having said that, please note that I'm reviewing each one and excepting someone on the other side to provide the same answers you just raised.
Without those answers a PR can't be merged.
Unfortunately, I don't have the time to take each branch and test it locally to see these kind of changes and I need to relay on the contributor to do the manual testing part.
In most cases I think it works well, but it's a brave new world now.
Please don't be demotivated by these PRs, you can treat them as a suggestion and decide if you want to review them, take their code and open a different PR with some of these changes or just disregard them.
We have an AI policy in place which you can read, which favors human interaction.
If you think the process should be improved ping me on slack and we can brainstorm on this.

**Comment by smmariquit** [comment]
opened #8045 for this.

default `draggable: true` markers get `tabindex="0"`. while focused, arrow keys move them 1 screen px per press (10 with Shift) and fire the same `dragstart`/`drag`/`dragend` gesture as pointer dragging: position updates first, `dragstart` once on the first movement, `drag` per movement, `dragend` on key release. handled arrow keys don't also pan the map. custom `element` markers are left alone per #7790.

road-tested in Chromium on `test/examples/create-a-draggable-marker.html`. GIF and screen-pixel measurements are on the PR.

#8022 is also open for this issue; happy to close mine if that direction is preferred.
